adds support for npc vending machines

Fikou opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

there's now npc vending machines, similar to regular ones, but are more human, so you know, you cant deconstruct them with tools
fixes a bug where the special vending phrase of a machine wasnt used at all and it was always default
moves face_atom to atom level
machine/structure examine now supports different genders

adds mr. bones as an example npc, you can sell bones and milk to him and buy some stuff, admin spawnable

jegub made the cool mr bones sprite thank him

video: https://streamable.com/cbg1h8
^ PSST, IGNORE THAT VIDEO, THIS ONE IS MUCH COOLER AND NEWER 
VVVVVVVVVVVVVVVVVVVVV
https://streamable.com/oqofto

## Why It's Good For The Game

these seem useful for mapping or admin events where you can buy some unique items

## Changelog
:cl: Fikou, jegub
add: Added support for NPC vendors
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Fikou committed:
vending npcs

Fikou committed:
part 2 of human vendors

Fikou committed:
speen

Fikou committed:
this actually works so its better

Ryll-Ryll commented:
But can they fall over on you?

Fikou commented:
> But can they fall over on you?

that would break rule 8...

davethwave commented:
This has the potential for great admemery and perhaps some nice space ruins and away missions.

ArcaneMusic commented:
"We've kept this leprechaun in the brig until we can buy his gold off of him, sir."
Excellent.

Pumpkinoe commented:
looks incredibly awesome!
one question though: did you check it's not possible to return clothing into the NPC's 'clothing slot' like a regular vendor?

Fikou committed:
it goes yah

it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes it goes GUILLOTINE

Fikou committed:
face me

Fikou committed:
hmgh

Fikou commented:
well the main part of pr is done but im gonna try tomorrow to make vending machine component and turn this into an actual mob so this is all nicer

optimumtact commented:
That is a terrible idea

bobbahbrown commented on code/modules/vending/npc.dm:
```diff
@@ -0,0 +1,150 @@
+			return
+		generate_cash(wanted_items[activehanditem.type], user)
+		return
+	else if (is_type_in_list(inactivehanditem, wanted_items))
+		generate_cash(wanted_items[inactivehanditem.type], user)
+		qdel(inactivehanditem)
+		return
```
what if the offhand is a stack

i would redesign this so that you have a proc ``proc_that_actually_sells_item(item)``, and have it return TRUE or FALSE on if it actually sold the item. check with active hand, if it returns FALSE then check with the item in the inactive hand

reduces code duplication and should improve legibility/maintainability

```
try_sell(user):
  if (!sell_item(active_item))
    sell_item(inactive_item)
```

bobbahbrown replied:
actually with the say statement below, you could even do something like

```
if (!(sell_item(active_item) || sell_item(inactive_item)))
  say(blah blah)
```

though selling the item in the offhand may be unexpected to the player, maybe have a confirmation that they want to sell it that also tells them how much they'll get paid for it?

bobbahbrown commented on code/modules/vending/npc.dm:
```diff
@@ -0,0 +1,150 @@
+//Vending NPC's, they are vendors that seem a bit more human, might be useful for mapping/admin events
```
Add DMdoc for all of your new procs too

bobbah reviewed (commented):
consider adding logging for transactions as well, it would be useful for seeing how players interact with the npcs as well as potentially for issues where players are abusing the npcs

Fikou commented:
> That is a terrible idea

turning it into a mob or the pr itself

Fikou committed:
Apply suggestions from code review

Co-authored-by: Bobbahbrown <bobbahbrown@gmail.com>

Fikou committed:
apple bottom jeans

boots with the fur

Fikou committed:
fixes

Fikou committed:
dm doc

Fikou committed:
bobbah said to add logging

Fikou committed:
oops

Fikou committed:
how did i delte this

Fikou committed:
preventing runtimes from badmins

Fikou committed:
sound

Fikou committed:
waagh 1

Fikou committed:
waaaaaaaaaaaaagh 2

Fikou committed:
more sense with stacks

Fikou commented:
ok its ready now i guess if oranges doesnt want it to be a mob because his message confused me

Fikou committed:
more modular i guess

Fikou committed:
Update npc.dm

Fikou committed:
why didnt i think of this before

Fikou committed:
uhhh

Fikou commented:
help

AnturK commented:
Alright, this is removing/excepting a lot of base vending functionality which makes it pretty hard to maintain later.
If you want this you'll need to instead abstract vending into a component that's applicable to mobs.
